server/game/Effects/Values/ConditionValue.js

```diff
@@ -1,13 +1,14 @@
 const EffectValue = require('./EffectValue');
 
 class ConditionValue extends EffectValue {
-    constructor(condition) {
+    constructor(condition, target = null) {
         super(condition);
         this.condition = condition || (() => true);
+        this.target = target;
     }
 
-    getValue() {
-        return this;
+    getValue(target) {
+        return new ConditionValue(this.condition, target);
```

Comment:
This can be expensive as we create a new ConditionValue for every target. Maybe just store the target?